### PR TITLE
fix: remove middleware restriction

### PR DESCRIPTION
**Pull request by ematipico**
#### What kind of changes does this PR include?

Removes some content from the middleware section.

#### Description

We plan to remove the restriction around `locals` being serialisable. 

Context: https://github.com/withastro/roadmap/pull/555#issuecomment-1557151100

PR: https://github.com/withastro/astro/pull/7174

### CODE REVIEW

File: src/content/docs/en/guides/middleware.mdx [hunk 1]
```diff
@@ -154,16 +154,14 @@ A function that intercepts (reads and modifies) the `Response` of a `Request` or
...
 
 `locals` is an object that lives and dies within a single Astro route; when your route page is rendered, `locals` won't exist anymore and a new one will be created. Information that needs to persist across multiple page requests must be stored elsewhere.
 
-In `dev` mode, Astro perform checks and will throw an error if the following conditions are not met:
-- `locals` needs to be serializable: The information must be serializable because storing information that evaluates at runtime is unsafe. If, for example, we were able to store a JavaScript function, an attacker could exploit the victim's website and execute some unsafe code.
-- `locals` can't be overridden: The value of locals needs to be an object, and it can't be overridden at runtime. Doing so would risk wiping out all the information stored by the user.
+In `dev` mode, Astro perform checks and will throw an error if `locals` are be overridden.
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/middleware.mdx:163]: Small typos!
```suggestion
In `dev` mode, Astro performs checks and will throw an error if `locals` are overridden.
```

File: src/content/docs/en/guides/middleware.mdx [hunk 2]
```diff
@@ -154,16 +154,14 @@ A function that intercepts (reads and modifies) the `Response` of a `Request` or
...
 
 `locals` is an object that lives and dies within a single Astro route; when your route page is rendered, `locals` won't exist anymore and a new one will be created. Information that needs to persist across multiple page requests must be stored elsewhere.
 
-In `dev` mode, Astro perform checks and will throw an error if the following conditions are not met:
-- `locals` needs to be serializable: The information must be serializable because storing information that evaluates at runtime is unsafe. If, for example, we were able to store a JavaScript function, an attacker could exploit the victim's website and execute some unsafe code.
-- `locals` can't be overridden: The value of locals needs to be an object, and it can't be overridden at runtime. Doing so would risk wiping out all the information stored by the user.
+In `dev` mode, Astro performs checks and will throw an error if `locals` are overridden.
+The value of locals needs to be an object, and it can't be overridden at runtime. Doing so would risk wiping out all the information stored by the user.
```
Comment (sarah11918) [inline src/content/docs/en/guides/middleware.mdx:164]:
```suggestion
:::note
The value of `locals` cannot be overridden at run time. Doing so would risk wiping out all the information stored by the user.  In `dev` mode, Astro performs checks and will throw an error if `locals` are overridden.
:::
```
What do we think about this, and making it a note? I feel like swapping the order makes more sense now that there are not two different things that throw an error. Rather, this section is now just about not overriding `locals`, and we back it up by saying there's gonna be an error if you do?

**Review by sarah11918 — COMMENTED** [review]
Thanks @ematipico! What do you think about something like my suggestion below? (Explained there.)?

File: src/content/docs/en/guides/middleware.mdx (hunk 2 above)
Comment (ematipico) [reply src/content/docs/en/guides/middleware.mdx:164]: Thank you Sarah for the great suggestion! ❤️

**Review by sarah11918 — APPROVED** [review]
One more change when viewing the preview of this page just for visual effect! 

I checked and it looks like the change has already been released, so these docs can update immediately? If not, please add the `merge-on-release` label so that I DO NOT merge as soon as I update branch again and wait for the checks.  :)

**Pull request merged**